pkg/controllers/gitrepository/controller.go

```diff
@@ -0,0 +1,280 @@
+package gitrepository
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/cnoe-io/idpbuilder/api/v1alpha1"
+	"github.com/cnoe-io/idpbuilder/globals"
+	"github.com/cnoe-io/idpbuilder/pkg/controllers/localbuild"
+	"github.com/cnoe-io/idpbuilder/pkg/util"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	DefaultBranchName = "main"
+	// DefaultGiteaUsername and DefaultGiteaPassword taken from gitea helm chart. https://gitea.com/gitea/helm-chart#gitea
+	DefaultGiteaUsername = "gitea_admin"
+	DefaultGiteaPassword = "r8sA8CPHD9!bt6d"
+	requeueTime          = time.Second * 30
+	gitCommitAuthorName  = "git-reconciler"
+	gitCommitAuthorEmail = "invalid@cnoe.io"
+)
+
+type GiteaClientFunc func(url string, options ...gitea.ClientOption) (GiteaClient, error)
+
+func NewGiteaClient(url string, options ...gitea.ClientOption) (GiteaClient, error) {
+	return gitea.NewClient(url, options...)
+}
```

Comment:
not used?

Reply:
This needs to be used. It's just an implementation of the controller itself.